From 018fa284a97e81028690bd0e521208e02a0eec00 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 4 Feb 2020 11:09:32 +0200
Subject: [PATCH 13/16] add statusbar icon spacing tuning (crDroid)

  * based on
  https://github.com/neon-os/packages_apps_Settings/commit/ba07105ed23396f47d78bc3867bcd112ff1584bd

	modified:   res/values/strings.xml
	modified:   res/xml/display_settings.xml
	modified:   src/com/android/settings/DisplaySettings.java
	modified:   src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
	renamed:    src/com/android/settings/development/OverlayCategoryPreferenceController.java -> src/com/android/settings/display/OverlayCategoryPreferenceController.java
---
 res/values/strings.xml                           |  4 +++-
 res/xml/display_settings.xml                     | 10 ++++++++++
 src/com/android/settings/DisplaySettings.java    |  3 +++
 ...EmulateDisplayCutoutPreferenceController.java |  3 ++-
 .../OverlayCategoryPreferenceController.java     | 16 +++-------------
 5 files changed, 21 insertions(+), 15 deletions(-)
 rename src/com/android/settings/{development => display}/OverlayCategoryPreferenceController.java (92%)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index f5bd6f69a3..9fcbdbcb7d 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -10074,7 +10074,9 @@
     <string name="notification_log_details_ranking_none">Ranking object doesn\'t contain this key.</string>
 
     <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the settings category for theme overlays. -->
-    <string name="theme_customization_category">Theming</string>
+    <string name="theme_customization_category">Statusbar Theming</string>
+    <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the setting which enables overlays to customize the      statusbar icons spacing (e.g. set 2dp paddings). -->
+    <string name="theme_customization_statusbar_title">Status Bar spacing</string>
     <!-- [CHAR_LIMIT=NONE] Developer Settings: Label for the option that turns off customizations for a given category.-->
     <string name="theme_customization_device_default">Device default</string>
 
diff --git a/res/xml/display_settings.xml b/res/xml/display_settings.xml
index e27a126d3e..c1becdfe38 100644
--- a/res/xml/display_settings.xml
+++ b/res/xml/display_settings.xml
@@ -198,4 +198,14 @@
         android:title="@string/display_vr_pref_title"
         android:fragment="com.android.settings.display.VrDisplayPreferencePicker" />
 
+    <PreferenceCategory
+        android:key="theme_customization_category"
+        android:title="@string/theme_customization_category"
+        android:order="1200"> <!-- Incremented by 100 from last order (i.e. 1100) -->
+
+        <ListPreference
+            android:key="android.theme.customization.statusbar"
+            android:title="@string/theme_customization_statusbar_title"/>
+    </PreferenceCategory>
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/DisplaySettings.java b/src/com/android/settings/DisplaySettings.java
index 7e9c6ace62..162510aa4c 100644
--- a/src/com/android/settings/DisplaySettings.java
+++ b/src/com/android/settings/DisplaySettings.java
@@ -36,6 +36,7 @@ import com.android.settings.display.TapToWakePreferenceController;
 import com.android.settings.display.ThemePreferenceController;
 import com.android.settings.display.TimeoutPreferenceController;
 import com.android.settings.display.VrDisplayPreferenceController;
+import com.android.settings.display.OverlayCategoryPreferenceController;
 import com.android.settings.search.BaseSearchIndexProvider;
 import com.android.settings.search.Indexable;
 import com.android.settingslib.core.AbstractPreferenceController;
@@ -121,6 +122,8 @@ public class DisplaySettings extends DashboardFragment {
         controllers.add(new ShowOperatorNamePreferenceController(context));
         controllers.add(new ThemePreferenceController(context));
         controllers.add(new BrightnessLevelPreferenceController(context, lifecycle));
+        controllers.add(new OverlayCategoryPreferenceController(context,
+                "android.theme.customization.statusbar"));
         return controllers;
     }
 
diff --git a/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java b/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
index ef88baa134..c4f7838bf3 100644
--- a/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
+++ b/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
@@ -22,6 +22,7 @@ import android.content.pm.PackageManager;
 import android.os.ServiceManager;
 import android.view.DisplayCutout;
 
+import com.android.settings.display.OverlayCategoryPreferenceController;
 import androidx.annotation.VisibleForTesting;
 
 public class EmulateDisplayCutoutPreferenceController extends OverlayCategoryPreferenceController {
@@ -31,7 +32,7 @@ public class EmulateDisplayCutoutPreferenceController extends OverlayCategoryPre
     @VisibleForTesting
     EmulateDisplayCutoutPreferenceController(Context context, PackageManager packageManager,
             IOverlayManager overlayManager) {
-        super(context, packageManager, overlayManager, DisplayCutout.EMULATION_OVERLAY_CATEGORY);
+        super(context, DisplayCutout.EMULATION_OVERLAY_CATEGORY);
     }
 
     public EmulateDisplayCutoutPreferenceController(Context context) {
diff --git a/src/com/android/settings/development/OverlayCategoryPreferenceController.java b/src/com/android/settings/display/OverlayCategoryPreferenceController.java
similarity index 92%
rename from src/com/android/settings/development/OverlayCategoryPreferenceController.java
rename to src/com/android/settings/display/OverlayCategoryPreferenceController.java
index 6e0b2d00f6..6a20e3aac3 100644
--- a/src/com/android/settings/development/OverlayCategoryPreferenceController.java
+++ b/src/com/android/settings/display/OverlayCategoryPreferenceController.java
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.android.settings.development;
+package com.android.settings.display;
 
 import static android.os.UserHandle.USER_SYSTEM;
 
@@ -35,8 +35,8 @@ import androidx.preference.Preference;
 import androidx.preference.PreferenceScreen;
 
 import com.android.settings.R;
+import com.android.settingslib.core.AbstractPreferenceController;
 import com.android.settings.core.PreferenceControllerMixin;
-import com.android.settingslib.development.DeveloperOptionsPreferenceController;
 
 import java.util.ArrayList;
 import java.util.Comparator;
@@ -47,7 +47,7 @@ import java.util.List;
  * The chosen overlay is enabled exclusively within its category. A default option is also
  * exposed that disables all overlays in the given category.
  */
-public class OverlayCategoryPreferenceController extends DeveloperOptionsPreferenceController
+public class OverlayCategoryPreferenceController extends AbstractPreferenceController
         implements Preference.OnPreferenceChangeListener, PreferenceControllerMixin {
     private static final String TAG = "OverlayCategoryPC";
     @VisibleForTesting
@@ -194,14 +194,4 @@ public class OverlayCategoryPreferenceController extends DeveloperOptionsPrefere
         filteredInfos.sort(OVERLAY_INFO_COMPARATOR);
         return filteredInfos;
     }
-
-    @Override
-    protected void onDeveloperOptionsSwitchDisabled() {
-        super.onDeveloperOptionsSwitchDisabled();
-        // TODO b/133222035: remove these developer settings when the
-        // Settings.Secure.THEME_CUSTOMIZATION_OVERLAY_PACKAGES setting is used
-        setOverlay(PACKAGE_DEVICE_DEFAULT);
-        updateState(mPreference);
-    }
-
 }
-- 
2.25.1

